Update create transaction example code

tcharding opened the pull request:
### Description

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

The transaction builder changed a while ago, looks like some of the example code did not get updated.

Update the transaction creation code to use a mutable builder.

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

tcharding committed:
Update create transaction example code

The transaction builder changed a while ago, looks like some of the
example code did not get updated.

Update the transaction creation code to use a mutable builder.

afilini commented:
Looks like they are being ignored for some reason, can you also change them to `no_run` instead so that we can keep them updated going forward?

I did a quick test and there's also another test failing on line 145 which can be trivially fixed (replace `None` with `SignOptions::default()`)

tcharding commented:
Thanks for the review, will fix as suggested.

tcharding committed:
Use no_run instead of ignore

We have an attribute `no_run` that builds but does not run example code
in Rustdocs, this keeps the examples building as the codebase evolves.

use `no_run` and fix example code so it builds cleanly during test run.

tcharding committed:
Use project import style in example code

Since we are cleaning up the example code in `lib.rs`; use the `bdk`
project imports stlye in the example code i.e., separate upstream
imports form bdk/crate imports.

tcharding commented:
@afilini I fixed as suggested including changing all `ignore`s in `lib.rs` to `no_run`s (with code fixes). And, for the awesomely anal among us, I fixed the import lines to use `dbk` project style.

afilini commented:
I think the CI is failing because the tests use the electrum stuff but they are not feature-gated. It's kinda ugly but there's a way to fix it:

https://github.com/bitcoindevkit/bdk/blob/12641b9e8fa13cbc5eb34f55d18b423afc505aad/src/keys/mod.rs#L453-L472

afilini commented:
Since we have to do the release today and this isn't ready yet I've cherry-picked abbea4b on the release branch, so at least we will have the correct example on the published crate

tcharding commented:
Wow, this opens a whole can of worms. The `lib.rs` inner doc comment documents behaviour from the default set of features (as one would expect) but we cannot expect folks to build with the default set of features so we do not currently build these docs. Attempting to use the idea suggested above leads to this error:
```
note: inner doc comments like this (starting with `//!` or `/*!`) can only appear before items
```
I don't have the inclination to dig further into this at the moment. Closing this issue for another day.
